src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -1137,12 +1137,24 @@ bool QgsPostgresConn::supportedLayersPrivate( QVector<QgsPostgresLayerProperty>
   return true;
 }
 
-bool QgsPostgresConn::getSchemas( QList<QgsPostgresSchemaProperty> &schemas )
+bool QgsPostgresConn::getSchemas( QList<QgsPostgresSchemaProperty> &schemas, const QStringList &restrictToSchemas )
```

Comment:
this function allows to take several schemas but the UI allows only one, should we not changing this function or the UI to be both consistent ?

Reply:
Just two cents regarding the UI.
1) We could have a combobox to choose one schema from the database instead of a text field.
2) We could have a button on the UI for "Schema selection". It opens a model dialog where the user can choose which schemas to consider, like this one in DBeaver:
![imagem](https://github.com/user-attachments/assets/17348113-1e78-4c20-b716-de2055b56f27)
We could show the tables in each schema to help the user decide.

This can be made in a another PR, since the underlying functionality is already provided by this one.

Reply:
@troopa81 
> this function allows to take several schemas but the UI allows only one, should we not changing this function or the UI to be both consistent ?

At some stage it would be good to extend this to allow selection of multiple schemas in the UI, like what we support for SQL Server connections. It's out of scope for my current work (which is just limited to matching the Oracle provider functionality), but I think it doesn't hurt to have the API ready for this.

Reply:
@jgrocha 
I agree, but that's out of scope for my current contract.